src/modules/flake-parts/implementation.nix

```diff
@@ -0,0 +1,49 @@
+{
+  config,
+  lib,
+  ...
+}: let
+  l = lib // builtins;
+  d2n = config.dream2nix;
+  makeArgs = p:
+    {
+      inherit (config) systems;
+      inherit (d2n) config;
+    }
+    // p;
+  outputs = d2n.lib.dlib.mergeFlakes (
+    l.map
+    (p: d2n.lib.makeFlakeOutputs (makeArgs p))
+    (l.attrValues d2n.inputs)
+  );
+in {
+  config = {
+    flake =
+      # make attrs default, so that users can override them without
+      # needing to use lib.mkOverride (usually, lib.mkForce)
+      l.mapAttrsRecursiveCond
+      d2n.lib.dlib.isNotDrvAttrs
+      (_: l.mkDefault)
+      outputs;
+    dream2nix.outputs = outputs;
+    perSystem = {
+      config,
+      system,
+      ...
+    }: let
+      # get output attrs that have systems
+      systemizedOutputs =
+        l.mapAttrs
+        (_: attrs: attrs.${system})
+        (
+          l.filterAttrs
+          (_: attrs: l.isAttrs attrs && l.hasAttr system attrs)
+          outputs
+        );
+    in {
+      config = {
+        dream2nix.outputs = systemizedOutputs;
+      };
+    };
```

Comment:
Why do we need the whole `perSystem` thing? As we already defined `flake`, isn't that redundant? Does it give the user any extra features?

Reply:
It gives convenience yes. If you are in a system context, (ie. `perSystem`) you will be able to use `config.dream2nix.outputs.packages.ripgrep` instead of `config.dream2nix.outputs.packages.${system}.ripgrep`. Which removes one nesting so it is worth it IMO.

Reply:
For comparison, other modules like haskell-flake and pre-commit-hooks-nix exist mostly at the `perSystem` level. Users may expect to define things there instead of at the top level.
By taking configuration from the top-level instead of from the `perSystem` module, you deny users the ability to reference `system` in their dream2nix config. I'm not familiar with dream2nix yet, so I can't tell if this is a huge loss.

Reply:
Going the `perSystem` route by default, would probably solve problems like: https://github.com/nix-community/dream2nix/issues/345

So as the `perSystem` logic is now provided by flake-parts, we could just remove all multi-system handling from the entire dream2nix and reduce complexity in places. What do you think @yusdacra

Reply:
I just made #366 for this! Please check it out.